.github/workflows/continous_integration.yml

```diff
@@ -33,27 +33,44 @@ jobs:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v3
 
-      # Set up the python versions
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+      # Set up the python versions and install conda dependencies
+      - name: Install conda dependencies
+        uses: conda-incubator/setup-miniconda@v2
         with:
           python-version: ${{ matrix.python-version }}
+          environment-file: conda_env.yml
+          activate-environment: flarestack
+          auto-activate-base: false
+
+      # Display the python version and conda info
+      - name: Display Python version and Conda info
+        shell: bash -el {0}
+        run: |
+            python --version
+            conda --version
+            conda config --show-sources
+            conda config --show
+            conda list
+            conda info -a
 
      # Setting up dependencies
       - name: Install dependencies
+        shell: bash -el {0}
         run: |
           python -m pip install --upgrade pip
           python -m pip install poetry
           python -m poetry install 
```

Comment:
I think that installing `poetry` from/to within same virtual environment of the package can incidentally work but is  discouraged in general.